src/rollup/plugins/externals.ts

```diff
@@ -64,7 +65,7 @@ export function externals (opts: NodeExternalsOptions): Plugin {
       const resolved = await this.resolve(originalId, importer, { ...options, skipSelf: true }) || { id }
 
       // Try resolving with mlly as fallback
-      if (!existsSync(resolved.id)) {
+      if (!isAbsolute(resolved.id) || !existsSync(resolved.id) || await isDirectory(resolved.id)) {
         resolved.id = await _resolve(resolved.id).catch(() => resolved.id)
```

Comment:
Calling resolver on non-absolute paths is probably also wrong (URL from _resolve is node_modules). Can we instead simply test `id` against npm package name regex and try resolving?

Reply:
What kind of regexp did you have in mind?

Reply:
Ah. The edge case is where the input does _not_ look like an absolute path but does exist in the FS, e.g. `express` when there is a folder called `express`. I think it is sufficient for the linked bug to check whether it is a directory or not.

But I think we shouldn't just check for directory. After all, we _require_ an absolute path and we really should re-resolve it if it isn't one at this point. If we get a result like `express/index.js` this may be a file on the disk but we can't import the file that way.